app/uk/gov/hmrc/thirdpartyapplication/services/GatekeeperService.scala

```diff
@@ -150,9 +149,9 @@ class GatekeeperService @Inject()(applicationRepository: ApplicationRepository,
 
   }
 
-  def deleteApplication(applicationId: UUID, request: DeleteApplicationRequest)(implicit hc: HeaderCarrier): Future[ApplicationStateChange] = {
+  def deleteApplication(applicationId: UUID, request: Option[DeleteApplicationRequest])(implicit hc: HeaderCarrier): Future[ApplicationStateChange] = {
     def audit(app: ApplicationData): Future[AuditResult] = {
-        auditGatekeeperAction(request.gatekeeperUserId, app, ApplicationDeleted, Map("requestedByEmailAddress" -> request.requestedByEmailAddress))
+        auditGatekeeperAction(request.get.gatekeeperUserId.toString, app, ApplicationDeleted, Map("requestedByEmailAddress" -> request.get.requestedByEmailAddress.toString))
```

Comment:
I am not sure how this .get is safe in this situation.  If this can only be called with a valid DeleteApplicationRequest could we change the signature?

Reply:
Yeah I'm going to change that one, as at that point there's always a value.

Reply:
I've made this change and resolved the conflicts with master